### DISCUSSION TITLE
Optimizing Tasking Manager: Exploring the Move to FastAPI

### DISCUSSION THREAD

**Discussion by kaditya97** [hotosm/tasking-manager#6212] [Ideas]
Some of the reasons for switching to FastAPI
- Faster performance: Utilizes ASGI server and type annotations for faster execution and better resource utilization
- Asynchronous requests
- Automatic data validation and documentation generation
- Dependency injection, reducing boilerplate code and simplifying development

For initial testing, load testing was done for /projects/{id} API. The outcome of this is below. It is just an initial test and may not be actual output. 
- Flask load testing result
![screencapture-localhost-8089-stats-report-2024-01-20-18_23_04](https://github.com/hotosm/tasking-manager/assets/50064160/99220ac8-1bbc-4299-8b0f-457353023c1f)

- FastAPI load testing result
![screencapture-localhost-8089-stats-report-2024-01-20-18_05_46](https://github.com/hotosm/tasking-manager/assets/50064160/87ddd024-ab3b-4961-8154-87784684e236)

**Comment by robsavoye** [comment]
The other advantage is other HOT projects are using FastAPI, so it enables better sharing of code.
